FloripaTrans/FTFindRoutesGoogleMapsViewController.m

```diff
@@ -0,0 +1,130 @@
+    
+    // Set up the map view
+    self.mapView.myLocationEnabled = NO;
+    self.mapView.mapType = kGMSTypeNormal;
+    self.mapView.settings.compassButton = YES;
+    self.mapView.settings.myLocationButton = NO;
+    
+    self.mapView.delegate = self;
+    
+    // Round the corner of the search button
+    [self.okButton roundViewCornerWithRadius:10];
+    
+    // Set up initial location for the map view
+    GMSCameraPosition *camera = [GMSCameraPosition cameraWithLatitude:-27.593500
+                                                            longitude:-48.558540
+                                                                 zoom:14];
+    self.mapView.camera = camera;
+    
+    // Instantiate the route marker
+    self.routeMarker = [[GMSMarker alloc] init];
+    
+    // Change the back button so it has the title Back
+    UIBarButtonItem *item = [[UIBarButtonItem alloc] initWithTitle:@"Back" style:UIBarButtonItemStylePlain target:self action:@selector(goBack)];
+    self.navigationItem.hidesBackButton = YES;
+    self.navigationItem.leftBarButtonItem = item;
+    
+}
+
+- (void)goBack {
+    [self.navigationController popViewControllerAnimated:YES];
+}
+
+- (IBAction)okTouchUp:(id)sender {
+    if(!self.routeMarker.map) {
+        [self alertMessage:@"Choose a location" withTilte:@"Location Selection"];
+        return;
+    }
+    
+    [[UIApplication sharedApplication] setNetworkActivityIndicatorVisible:YES];
+    [self reverseGeocodeCoordinate:self.routeMarker.position];
+}
+
+
+/**
+ *  Method that check the content address based on a coordinate. If there's no error, the method calls a validation method to see if the address is a street or avenue.
+ *
+ *  @param coordinate coordinate
+ */
+- (void)reverseGeocodeCoordinate:(CLLocationCoordinate2D)coordinate {
+    GMSGeocoder *geocoder = [GMSGeocoder geocoder];
+    
+    [geocoder reverseGeocodeCoordinate:coordinate completionHandler:^(GMSReverseGeocodeResponse *response, NSError *error) {
+        
+        [[UIApplication sharedApplication] setNetworkActivityIndicatorVisible:NO];
+        
+        if(error) {
+            [self alertMessage:[error localizedDescription] withTilte:@"Map Error"];
+            
+            return;
+        }
+        
+        GMSAddress *address = [response firstResult];
+        if(address){
+           
+            [self checkForValidStreet:address.thoroughfare];
+         }
+    }];
+}
+
+/**
+ *  Check if the string is a street or an avenue. If it is valid, removes the AV. or R., call the delegate and pop the map view from the navigation stack. If it is not valid, it retrieves an erro message
+ *
+ *  @param string string which will be evaluated
+ */
+- (void) checkForValidStreet:(NSString *)string{
+    
+    NSError *error = nil;
+    NSRegularExpression *regex = [NSRegularExpression regularExpressionWithPattern:@"(.*Av\\.\\s)|(.*R\\.\\s)" options:NSRegularExpressionCaseInsensitive error:&error];
```

Comment:
Oh, I see. Because this was my first time using the Google Maps SDK, I didn't know what to expect from it. I've been searching here and found "Servidão" and "Travessa" that I'm missing. Is there any other that I might be missing?

Also I'm opening an issue to add these two that I mentioned above and will try to fix it right away.

Thanks in advance.